lib/liquid/traversal.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+module Liquid
+  class Traversal
```

Comment:
Visitor is a name that carries more useful connotation for me here. Is it the wrong name for a reason I'm missing?

Reply:
I felt like since this is just one possible instance of the visitor pattern, taking the name of the whole pattern for this case was over-broad.  But happy to call it `Visitor` if that makes more sense to you.

Reply:
`AllTypesVisitor`? I'm basically looking for the ability to have a decent idea what this does without reading the implementation.